aten/src/ATen/functorch/DynamicLayer.cpp

```diff
@@ -499,6 +499,37 @@ TORCH_LIBRARY_IMPL(_, FuncTorchDynamicLayerBackMode, m) {
   m.fallback(torch::CppFunction::makeFromBoxedFunction<&dynamicLayerBackFallback>());
 }
 
+TORCH_LIBRARY_IMPL(aten, FuncTorchDynamicLayerFrontMode, m) {
```

Comment:
Let me think about it some more. Morally, we shouldn't do the fallthrough business, because one day functorch will just be implemented using pre-dispatch modes and it is possible for each pre-dispatch mode to have their own custom sizes/strides behavior and we don't want to skip all of that logic.

The problem is that functorch can't do a re-dispatch inside of DynamicLayerFrontMode (which is happening because we try to construct a new TensorWrapper, which calls .sym_storage_offset, which is triggering the re-dispatch). So I want to see if we can figure out how to get that to work first

Reply:
The correct behavior should be: when we are constructing TensorWrapper, we should call .sym_storage_offset at the next mode on the functorch stack.

To get there, I think we just need to ExcludeDispatchKeyGuard on DynamicLayerFrontMode here: https://github.com/pytorch/pytorch/blob/e19a855b4ddead54c26b58d8aa5f8442310bea5e/aten/src/ATen/functorch/ADInterpreters.cpp#L31 instead of doing the fallthrough business

Reply:
updated the PR to do as you described (this works)